### ISSUE TITLE
Fails when used with an UploadField

### ISSUE THREAD

**Issue by nyeholt** [sheadawson/silverstripe-quickaddnew#27]
If the object you're quickadding has an upload field, addition of that object fails due to the form submission using an old version of jquery.form.js. (2.36)

https://github.com/sheadawson/silverstripe-quickaddnew/blob/master/code/extensions/QuickAddNewExtension.php#L70

This version creates a fake xhr object that looks like

```
var xhr = { // mock object
            aborted: 0,
            responseText: null,
            responseXML: null,
            status: 0,
            statusText: 'n/a',
            getAllResponseHeaders: function() {},
            getResponseHeader: function() {},
            setRequestHeader: function() {},
            abort: function() {
                this.aborted = 1;
                $io.attr('src', opts.iframeSrc); // abort op in progress
            }
        };
```

that doesn't define a `success` method; the cms' jquery.ondemand functionality uses .success for doing things, but as this fake object doesn't have it, the ondemand fails and so does everything else :) 

Changing to use `Requirements::javascript(THIRDPARTY_DIR . '/jquery-ui/jquery-ui.js');` works fine, but I'm not sure if there's a specific reason for using the jquery.validate version?

**Comment by sheadawson** [comment]
Hey too long ago to remember why that form.js is being used. As long as it works with the suggested update with and without validation on the quick add new form, happy to merge a PR :)

**Comment by sheadawson** [comment]
Fixed in #41
